packages/cdktf/lib/hcl/render.ts

```diff
@@ -15,6 +15,14 @@ function escapeQuotes(str: string): string {
   return str.replace(/(?<!\\)"/g, '\\"');
 }
 
+function wrapIdentifierInQuotesIfNeeded(key: string): string {
+  return /^(?:\d|[\-])/.test(key)
```

Comment:
Nit picking, but can't these be combined? Also wouldn't it be easier to test if a key is an identifier by using the logic [described in the docs](https://developer.hashicorp.com/terraform/language/syntax/configuration#identifiers) and quote in all other cases?

```suggestion
  return (/^(?:\d|[\-])/.test(key) || /[^A-Za-z0-9_\-]/.test(key)) ? `"${key}"` : key
```

Reply:
Yeah, I think you're right. I just got caught in the middle of my thought process where I didn't want to use `A-Za-z` detection because of unicode. I didn't find groups to use that were more unicode friendly, so ended up using it anyway and didn't further reduce everything.